src/java/org/pantsbuild/tools/junit/impl/ConsoleRunnerImpl.java

```diff
@@ -533,26 +533,32 @@ private int runLegacy(Collection<Spec> parsedTests, JUnitCore core) throws Initi
     if (this.parallelThreads > 1) {
       ConcurrentCompositeRequestRunner concurrentRunner = new ConcurrentCompositeRequestRunner(
           requests, this.defaultConcurrency, this.parallelThreads);
-      if (failFast) {
-        return core.run(new FailFastRunner(concurrentRunner)).getFailureCount();
-      } else {
-        return core.run(concurrentRunner).getFailureCount();
-      }
+      Runner runner = maybeWithFailFastRunner(concurrentRunner);
+      return core.run(runner).getFailureCount();
     }
 
     int failures = 0;
     Result result;
     for (Request request : requests) {
-      if (failFast) {
-        result = core.run(new FailFastRunner(request.getRunner()));
-      } else {
-        result = core.run(request);
-      }
+      result = core.run(runnerFor(request));
```

Comment:
This changes behaviour; is that intentional, or should this be a call to `maybeWithFailFastRunner`?

Reply:
Hm. I'd expected it to not change behavior. Are you referring to the fact that in the non-failfast case, it calls `run(Runner r)` instead of `run(Request r)`?  The Request based run just calls the Runner one, with the request's runner.
https://github.com/junit-team/junit4/blob/master/src/main/java/org/junit/runner/JUnitCore.java#L108-L116

Reply:
Oh, I was totally misreading, ignore me!